db/error_handler.cc

```diff
@@ -418,15 +418,18 @@ const Status& ErrorHandler::SetBGError(const IOStatus& bg_io_err,
                                           &bg_err, db_mutex_, &auto_recovery);
     recover_context_ = context;
     return bg_error_;
-  } else if (bg_io_err.GetScope() ==
-                 IOStatus::IOErrorScope::kIOErrorScopeFile ||
-             bg_io_err.GetRetryable()) {
+  } else if (bg_io_err.subcode() != IOStatus::SubCode::kNoSpace &&
```

Comment:
Interesting. Just realized that the ```==``` operator of ```Status``` and ```IOStatus``` only checks code and not subcode. So is there a potential bug in line 330?

Reply:
I will check line 330. I just realized the == operator only compares the code not subcode when did the implementation